Project/Assets/ML-Agents/Examples/SharedAssets/Scripts/TargetController.cs

```diff
@@ -0,0 +1,143 @@
+        public TriggerEvent onTriggerStayEvent = new TriggerEvent();
+        public TriggerEvent onTriggerExitEvent = new TriggerEvent();
+
+        [System.Serializable]
+        public class CollisionEvent : UnityEvent<Collision>
+        {
+        }
+
+        [Header("Collision Callbacks")] public bool colliderIsTouching;
+        public CollisionEvent onCollisionEnterEvent = new CollisionEvent();
+        public CollisionEvent onCollisionStayEvent = new CollisionEvent();
+        public CollisionEvent onCollisionExitEvent = new CollisionEvent();
+
+        // Start is called before the first frame update
+        void OnEnable()
+        {
+            m_startingPos = transform.position;
+            if (respawnIfTouched)
+            {
+                MoveTargetToRandomPosition();
+            }
+        }
+
+        void Update()
+        {
+            if (respawnIfFallsOffPlatform)
+            {
+                if (transform.position.y < m_startingPos.y - fallDistance)
+                {
+                    Debug.LogError($"{transform.name} Fell Off Platform");
+                    MoveTargetToRandomPosition();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Moves target to a random position within specified radius.
+        /// </summary>
+        public void MoveTargetToRandomPosition()
+        {
+            var newTargetPos = m_startingPos + (Random.insideUnitSphere * spawnRadius);
+            newTargetPos.y = 5;
+            transform.position = newTargetPos;
+            triggerIsTouching = false;
+            colliderIsTouching = false;
+        }
+
+        private void OnCollisionEnter(Collision col)
+        {
+            if (col.transform.CompareTag(tagToDetect))
+            {
+                colliderIsTouching = true;
+                onCollisionEnterEvent.Invoke(col);
+                if (respawnIfTouched)
+                {
+                    MoveTargetToRandomPosition();
+                }
+            }
+        }
+
+        private void OnCollisionStay(Collision col)
+        {
+            if (col.transform.CompareTag(tagToDetect))
+            {
+                colliderIsTouching = true;
+                onCollisionStayEvent.Invoke(col);
+            }
+        }
+
+        private void OnCollisionExit(Collision col)
+        {
+            if (col.transform.CompareTag(tagToDetect))
+            {
+                colliderIsTouching = false;
+                onCollisionExitEvent.Invoke(col);
+            }
+        }
+
+        private void OnTriggerEnter(Collider col)
+        {
+            if (col.CompareTag(tagToDetect))
+            {
+                triggerIsTouching = true;
+                onTriggerEnterEvent.Invoke(col);
+            }
+        }
+
+        private void OnTriggerStay(Collider col)
+        {
+            if (col.CompareTag(tagToDetect))
+            {
+                triggerIsTouching = true;
```

Comment:
`triggerIsTouching` and `colliderIsTouching` are both bools that can be set in the inspector, but it seems they aren't actually used for anything. They are just turned on and off. Can we remove them?

Reply:
yeah they're not currently used. was more for potential future use. I can remove.

Reply:
changed. i removed both properties. will add in the future if needed